src/settings.c

```diff
@@ -37,54 +72,44 @@ static GQueue *xdg_config() {
+                                        ? xdg_cdirs
+                                        : g_strdup(XDG_CONFIG_DIRS_DEFAULT);
+        g_free(xdg_cdirs);
 
         /*
          * Prepend XDG_CONFIG_HOME, most important first because XDG_CONFIG_DIRS
          * is already ordered that way.
          */
         gchar * const all_conf_dirs = g_strconcat(g_get_user_config_dir(), ":",
-                                                  xdg_config_dirs, NULL);
+                                                  g_strdup(xdg_config_dirs), NULL);
+        g_free(xdg_config_dirs);
         LOG_D("Config directories: '%s'", all_conf_dirs);
 
-        FILE *f = NULL;
-
-        /*
-         * Possible relative paths to dunstrc. First match wins.
-         * TODO: Maybe warn about deprecated location if found?
-         */
-        const char * const rel_paths[] = { "dunst/dunstrc",
-                                           "dunstrc", // deprecated since v0.2 (2013-06-23)
-                                           NULL };
-
-        // Used as stack, least important pushed last but popped first.
-        GQueue *config_files = g_queue_new();
-        for (gchar * const *d = string_to_array(all_conf_dirs, ":"); *d; d++)
-                // FIXME: Flatten this out after deprecation grace period
-                for (const char * const *rp = rel_paths; *rp; rp++) {
-                        f = fopen(string_to_path(g_strconcat(*d, "/", *rp, NULL)), "r");
-                        LOG_I("Trying to open config file '%s' in '%s': '%s'",
-                                                          *rp,
-                                                                   *d,
-                                                                         f ? "HIT" : "MISS");
-                        if (f) {
-                                g_queue_push_tail(config_files, f);
-                                break; // ignore deprecated
-                        }
-                }
+        for (gchar * const *d = string_to_array(all_conf_dirs, ":"); *d; d++) {
+                gchar * const path = string_to_path(g_strconcat(*d, "/", RPATH_RC, NULL));
+                if ((config_file = fopen(path, "r"))) {
+                        LOG_I(MSG_FOPEN_SUCCESS(path, config_file));
+                        g_queue_push_tail(config_files, config_file);
+                } else
+                        // debug level because of low relevance
+                        LOG_D(MSG_FOPEN_FAILURE(path));
+                g_free(path);
+        }
         g_free(all_conf_dirs);
 
         return config_files;
 }
 
 void settings_init() {
-        static int count = 0;
-        if (count == 0)
+        static bool init_done = false;
+        if (!init_done) {
+                LOG_D("Initializing settings");
                 settings = (struct settings) {0};
-        count++;
+
+                init_done = true;
+        }
 }
```

Comment:
I could not make sense of the counter, a bool latch seems more appropriate.

Reply:
Yup